syntax = "proto3";

package reporting;

enum Alignment {
  ALLY = 0;
  ENEMY = 1;
}

enum ShipClass {
  CORVETTE = 0;
  FRIGATE = 1;
  CRUISER = 2;
  DESTROYER = 3;
  CARRIER = 4;
  DREADNOUGHT = 5;
}

message Officer {
  string first_name = 1;
  string last_name = 2;
  string rank = 3;
}

message Spaceship {
  optional Alignment alignment = 1;
  string name = 2;
  optional ShipClass ship_class = 3;
  float length = 4;
  int32 crew_size = 5;
  optional bool is_armed = 6;
  repeated Officer officers = 7;
}

service Reporting {
  rpc GetSpaceshipStream (EclipticCoordinates) returns (stream Spaceship) {}
}

message EclipticCoordinates {
  float longitude = 1;
  float latitude = 2;
  float distance = 3;
}